test/FlatLaunchpeg.test.ts

```diff
@@ -63,28 +63,36 @@ describe('FlatLaunchpeg', () => {
   beforeEach(async () => {
     config = { ...(await getDefaultLaunchpegConfig()) }
     await deployFlatLaunchpeg()
-    await flatLaunchpeg.setPublicSaleActive(true)
   })
 
   describe('Initialization', () => {
+    it('Phases can be initialized only once', async () => {
+      await initializePhasesFlatLaunchpeg(flatLaunchpeg, config, Phase.NotStarted)
+      await expect(initializePhasesFlatLaunchpeg(flatLaunchpeg, config, Phase.DutchAuction)).to.be.revertedWith(
+        'Launchpeg__AuctionAlreadyInitialized()'
+      )
+    })
+
+    it('Sale dates should be correct', async () => {
+      config.allowlistStartTime = BigNumber.from(0)
+      await expect(initializePhasesFlatLaunchpeg(flatLaunchpeg, config, Phase.DutchAuction)).to.be.revertedWith(
+        'Launchpeg__InvalidStartTime()'
+      )
+    })
+
+    it('Public sale must happen after allowlist', async () => {
+      config.publicSaleStartTime = config.auctionStartTime.sub(duration.minutes(20))
```

Comment:
might be missing something, can we not use auctionStartTime here?

```suggestion
      config.publicSaleStartTime = config.allowlistStartTime.sub(duration.minutes(20))
```

Reply:
I fixed all the "auction" things coming from copy-pasting